src/plugins/mer/merbuildconfiguration.cpp

```diff
@@ -50,11 +56,25 @@ namespace Internal {
 
 MerBuildConfiguration::MerBuildConfiguration(Target *target, Core::Id id)
     : QmakeBuildConfiguration(target, id)
-    , m_useExtraParserArguments(false)
+    , m_useExtraParserArguments(MerSettings::isImportQmakeVariablesEnabled())
 {
     QmakeProject *qmakeProject = static_cast<QmakeProject *>(project());
     connect(qmakeProject, &QmakeProject::buildDirectoryInitialized,
             this, &MerBuildConfiguration::setupExtraParserArguments);
+
+    connect(EditorManager::instance(), &EditorManager::aboutToSave,
+            this, [this](IDocument *document) {
+        QTC_ASSERT(project(), return);
+        if (!project()->files(Project::AllFiles).contains(document->filePath())
+                || !document->filePath().toString().contains(QRegExp("\\.spec$|\\.yaml$"))
+                || !m_useExtraParserArguments)
+            return;
+
+        updateExtraParserArguments();
+    });
```

Comment:
This code should go to the earlier commit. This commit should only deal with the checkbox addition.

Reply:
Fixed.